core/templates/dev/head/pages/collection_editor/settings_tab/CollectionPermissionsCardDirective.js

```diff
@@ -22,10 +22,24 @@ oppia.directive('collectionPermissionsCard', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_permissions_card_directive',
     controller: [
-      '$scope',
-      function($scope) {
-        $scope.ownerNames = GLOBALS.ownerNames;
-        $scope.isPrivate = GLOBALS.isPrivate;
+      '$scope', 'CollectionRightsObjectFactory',
+      'CollectionRightsBackendApiService',
+      function($scope, CollectionRightsObjectFactory,
+        CollectionRightsBackendApiService) {
+        $scope.collectionId = GLOBALS.collectionId;
+        $scope.collectionRightsObject = null;
+        $scope.hasRightsLoaded = false;
+        $scope.isPrivate =
```

Comment:
That's odd. Are you sure it's a race condition? JavaScript doesn't utilize multiple threads (at least, not in a way where it should produce race conditions). Something else seems to be wrong, though I'm not clear what.

The solution I suggested is used by the collections object and we haven't had consistency issues with it. Have you tried a similar solution?